From 4bdd8994d43dc30013d4e1e607ec268413c60ec4 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:26:49 -0400
Subject: [tools] perf: Compare hists comm by addresses

Message-id: <1408462094-14194-241-git-send-email-jolsa@redhat.com>
Patchwork-id: 88255
O-Subject: [PATCH RHEL7.1 BZ1131394 240/325] perf tools: Compare hists comm by addresses
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit fedd63d3cdc9004df43b02df5c874b8957992fe8
Author: Frederic Weisbecker <fweisbec@gmail.com>
Date: Wed Sep 11 17:18:09 2013 +0200

description
===========
Now that comm strings are allocated only once and refcounted to be shared
among threads, these can now be safely compared by addresses. This
should remove most hists collapses on post processing.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/sort.c b/tools/perf/util/sort.c
index 835e8bd..bf91d0e 100644
--- a/tools/perf/util/sort.c
+++ b/tools/perf/util/sort.c
@@ -80,7 +80,8 @@ struct sort_entry sort_thread = {
 static int64_t
 sort__comm_cmp(struct hist_entry *left, struct hist_entry *right)
 {
- return right->thread->tid - left->thread->tid;
+ /* Compare the addr that should be unique among comm */
+ return thread__comm_str(right->thread) - thread__comm_str(left->thread);
 }
 
 static int64_t
-- 
1.7.1